### PR TITLE
Add specific memory backend for virtiofs test case on s390x

### PR DESCRIPTION
**Pull request by fbq815**
Virtio_fs: Add several limination for virtiofs and remove numa related
cases on s390x

ID: 1500
    
Signed-off-by: Boqiao Fu <bfu@redhat.com>

### CODE REVIEW

**Comment by fbq815** [comment]
Test result:
RHEL.8:
JOB ID     : c44fcda9d9b9cf44973a147af17631543579fe20
JOB LOG    : /root/avocado/job-results/job-2024-04-19T05.56-c44fcda/job.log
 (1/3) Host_RHEL.m8.u10.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.10.0.s390x.io-github-autotest-qemu.virtio_fs_share_data_hugepage.basic_test.with_cache.auto.s390-virtio: STARTED
 (1/3) Host_RHEL.m8.u10.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.10.0.s390x.io-github-autotest-qemu.virtio_fs_share_data_hugepage.basic_test.with_cache.auto.s390-virtio: PASS (47.42 s)
 (2/3) Host_RHEL.m8.u10.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.10.0.s390x.io-github-autotest-qemu.virtio_fs_share_data_hugepage.basic_test.with_cache.always.s390-virtio: STARTED
 (2/3) Host_RHEL.m8.u10.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.10.0.s390x.io-github-autotest-qemu.virtio_fs_share_data_hugepage.basic_test.with_cache.always.s390-virtio: PASS (46.68 s)
 (3/3) Host_RHEL.m8.u10.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.10.0.s390x.io-github-autotest-qemu.virtio_fs_share_data_hugepage.basic_test.with_cache.none.s390-virtio: STARTED
 (3/3) Host_RHEL.m8.u10.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.10.0.s390x.io-github-autotest-qemu.virtio_fs_share_data_hugepage.basic_test.with_cache.none.s390-virtio: PASS (49.03 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2024-04-19T05.56-c44fcda/results.html
JOB TIME   : 145.21 s

RHEL.9:
 (1/4) Host_RHEL.m9.u4.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.10.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: STARTED
 (1/4) Host_RHEL.m9.u4.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.10.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio:PASS (506.54 s)
 (2/4) Host_RHEL.m9.u4.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.10.0.s390x.io-github-autotest-qemu.balloon_in_use.with_stop_continue.during_bg_test.s390-virtio: STARTED
 (2/4) Host_RHEL.m9.u4.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.10.0.s390x.io-github-autotest-qemu.balloon_in_use.with_stop_continue.during_bg_test.s390-virtio: PASS (18453.27 s)
 (3/4) Host_RHEL.m9.u4.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.10.0.s390x.io-github-autotest-qemu.balloon_in_use.with_reboot.during_bg_test.s390-virtio: STARTED
 (3/4) Host_RHEL.m9.u4.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.10.0.s390x.io-github-autotest-qemu.balloon_in_use.with_reboot.during_bg_test.s390-virtio: PASS (11820.79 s)
 (4/4) Host_RHEL.m9.u4.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.10.0.s390x.io-github-autotest-qemu.balloon_in_use.with_live_migration.during_bg_test.s390-virtio: STARTED
 (4/4) Host_RHEL.m9.u4.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.10.0.s390x.io-github-autotest-qemu.balloon_in_use.with_live_migration.during_bg_test.s390-virtio: PASS (16105.70 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2024-04-19T03.08-92beebb/results.html
JOB TIME   : 46889.58 s

**Comment by fbq815** [comment]
@PaulYuuu please help me review this patch

File: qemu/tests/cfg/virtio_fs_share_data_hugepage.cfg [hunk 1]
```diff
@@ -5,6 +5,15 @@
...
     required_qemu = [4.2.0,)
+    s390, s390x:
+        required_qemu = [5.2.0,)
+        vm_mem_share = yes
+        pre_command_noncritical = yes
+        pre_command = "echo 3 > /proc/sys/vm/drop_caches"
+        setup_hugepages = yes
+        kvm_module_parameters = 'hpage=1'
+        expected_hugepage_size = 1024
+        cmd_dd = 'dd if=/dev/urandom of=%s bs=1M count=200 oflag=direct'
```
Comment (PaulYuuu) [inline qemu/tests/cfg/virtio_fs_share_data_hugepage.cfg:16]: It will be overridden by L66.

**Comment by fbq815** [comment]
Test result:
RHEL8:
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2024-05-06T23.28-cc59141/results.html
JOB TIME   : 94.10 s
RHEL9:
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2024-05-06T23.28-b65ddc3/results.html
JOB TIME   : 90.69 s
@PaulYuuu please help me review and merge, thx

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**